### PR TITLE
API refactor

### PR DESCRIPTION
**Pull request by daytoncf**
I refactored part of the `/grade` handler to make the code more readable and reduce some repetition. As part of this refactor I added a type `DbAssignment` and a function `GetAssignment` to the `db` package to keep our codebase consistent.

### CODE REVIEW

**Comment by daytoncf** [comment]
I used `git merge` before making the PR, hence the many commits. It's actually only like 3 or 4 commits. Not sure if there was a better way to do that lol.

File: api.go [hunk 1]
```diff
@@ -131,8 +98,8 @@ func Server() {
 			}
 
 			// Prepare request object to send files from form to bucket
-			bucketUrl := "https://kasxttiggmakvprevgrp.supabase.co/storage/v1/object/assignments/" + assignmentId + "/"
-
+			bucketUrl := supabase.BaseURL + "/" + "storage/v1/object/assignments/" + assignmentId + "/"
```
Comment (jacksonwelsh) [inline api.go:101]: nit- can probably combine the `"/"` and `"storage[...]"`

**Review by jacksonwelsh — APPROVED** [review]
lgtm you can probably get smaller diffs by running `git rebase main` instead of `git merge`

File: api.go (hunk 1 above)
Comment (daytoncf) [reply api.go:101]: cant believe ive done this...

**Pull request merged**